lib/src/tree_cursor.c

```diff
@@ -253,7 +256,11 @@ bool ts_tree_cursor_goto_last_child(TSTreeCursor *self) {
         return false;
     }
   }
-  return false;
+#ifdef _MSC_VER
+  __assume(0);
+#else
+  __builtin_unreachable();
+#endif
```

Comment:
Can we revert this and ignore the warning MSVC unnecessarily complains about?

Reply:
I mean this is technically an optimisation.

Reply:
a compiler from this millennia would know that this is unreachable code regardless